[CI] MultiClusterSpecIT class failing

**Build Scans:**
- [elasticsearch-periodic #4088 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/ugssigzo3reua)
- [elasticsearch-periodic #4088 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/czua3adjf646g)
- [elasticsearch-periodic #4084 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/a5gqylksso6sa)
- [elasticsearch-periodic #4084 / 8.13.4_bwc](https://gradle-enterprise.elastic.co/s/ubwsiunwaqvpy)

**Reproduction Line:**
```
./gradlew ':x-pack:plugin:esql:qa:server:multi-clusters:v8.14.3#bwcTest' -Dtests.class="org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT" -Dtests.method="test {enrich.SpatialEnrichmentKeywordMatchAndSpatialAggregation#[skip:-8.13.99, reason:st_intersects added in 8.14]}" -Dtests.seed=9E1D11D20C2FBDBB -Dtests.bwc=true -Dtests.locale=nyn-Latn-UG -Dtests.timezone=Asia/Ujung_Pandang -Druntime.java=22
```

**Applicable branches:**
8.15

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
org.elasticsearch.client.ResponseException: method [POST], host [http://[::1]:42841], URI [/_query?pretty=true&error_trace=true], status line [HTTP/1.1 500 Internal Server Error]
---
error:
  root_cause:
  - type: "transport_serialization_exception"
    reason: "Failed to deserialize response from handler [ContextRestoreResponseHandler[ActionListenerResponseHandler[refCounting[org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/WrappedActionListener{org.elasticsearch.xpack.esql.execution.PlanExecutor$$Lambda/0x00007f6ec85a16b8@2149c9cd}{org.elasticsearch.xpack.esql.execution.PlanExecutor$$Lambda/0x00007f6ec85a18e8@72437f48}/org.elasticsearch.xpack.esql.session.EsqlSession$$Lambda/0x00007f6ec85dd548@5c0b27a/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@58067d8d/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@376ed0e1/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@5e3e507b/org.elasticsearch.xpack.esql.session.EsqlSession$$Lambda/0x00007f6ec85df790@5eed7aa4/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener$$Lambda/0x00007f6ec81f6ad8@356c8ee1][org.elasticsearch.xpack.esql.enrich.EnrichPolicyResolver$$Lambda/0x00007f6ec86564d0@74c4b975]]]]"
    stack_trace: "org.elasticsearch.transport.TransportSerializationException: Failed\
      \ to deserialize response from handler [ContextRestoreResponseHandler[ActionListenerResponseHandler[refCounting[org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/WrappedActionListener{org.elasticsearch.xpack.esql.execution.PlanExecutor$$Lambda/0x00007f6ec85a16b8@2149c9cd}{org.elasticsearch.xpack.esql.execution.PlanExecutor$$Lambda/0x00007f6ec85a18e8@72437f48}/org.elasticsearch.xpack.esql.session.EsqlSession$$Lambda/0x00007f6ec85dd548@5c0b27a/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@58067d8d/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@376ed0e1/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@5e3e507b/org.elasticsearch.xpack.esql.session.EsqlSession$$Lambda/0x00007f6ec85df790@5eed7aa4/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener$$Lambda/0x00007f6ec81f6ad8@356c8ee1][org.elasticsearch.xpack.esql.enrich.EnrichPolicyResolver$$Lambda/0x00007f6ec86564d0@74c4b975]]]]\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.InboundHandler.doHandleResponse(InboundHandler.java:421)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.InboundHandler$2.doRun(InboundHandler.java:389)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\
      \tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\
      \tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\
      \tat java.base/java.lang.Thread.run(Thread.java:1570)\nCaused by: java.lang.IllegalArgumentException:\
      \ Invalid EsField type [airport]\n\tat org.elasticsearch.xpack.esql.core.type.EsField.getReader(EsField.java:38)\n\
      \tat org.elasticsearch.xpack.esql.io.stream.PlanStreamInput.readEsFieldWithCache(PlanStreamInput.java:280)\n\
      \tat org.elasticsearch.xpack.esql.core.type.EsField.readFrom(EsField.java:70)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:737)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:714)\n\
      \tat org.elasticsearch.xpack.esql.enrich.ResolvedEnrichPolicy.<init>(ResolvedEnrichPolicy.java:32)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:737)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:721)\n\
      \tat org.elasticsearch.xpack.esql.enrich.EnrichPolicyResolver$LookupResponse.<init>(EnrichPolicyResolver.java:333)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:63)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:25)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1471)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1458)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.InboundHandler.doHandleResponse(InboundHandler.java:418)\n\
      \t... 8 more\n"
  type: "transport_serialization_exception"
  reason: "Failed to deserialize response from handler [ContextRestoreResponseHandler[ActionListenerResponseHandler[refCounting[org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/WrappedActionListener{org.elasticsearch.xpack.esql.execution.PlanExecutor$$Lambda/0x00007f6ec85a16b8@2149c9cd}{org.elasticsearch.xpack.esql.execution.PlanExecutor$$Lambda/0x00007f6ec85a18e8@72437f48}/org.elasticsearch.xpack.esql.session.EsqlSession$$Lambda/0x00007f6ec85dd548@5c0b27a/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@58067d8d/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@376ed0e1/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@5e3e507b/org.elasticsearch.xpack.esql.session.EsqlSession$$Lambda/0x00007f6ec85df790@5eed7aa4/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener$$Lambda/0x00007f6ec81f6ad8@356c8ee1][org.elasticsearch.xpack.esql.enrich.EnrichPolicyResolver$$Lambda/0x00007f6ec86564d0@74c4b975]]]]"
  caused_by:
    type: "illegal_argument_exception"
    reason: "Invalid EsField type [airport]"
    stack_trace: "java.lang.IllegalArgumentException: Invalid EsField type [airport]\n\
      \tat org.elasticsearch.xpack.esql.core.type.EsField.getReader(EsField.java:38)\n\
      \tat org.elasticsearch.xpack.esql.io.stream.PlanStreamInput.readEsFieldWithCache(PlanStreamInput.java:280)\n\
      \tat org.elasticsearch.xpack.esql.core.type.EsField.readFrom(EsField.java:70)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:737)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:714)\n\
      \tat org.elasticsearch.xpack.esql.enrich.ResolvedEnrichPolicy.<init>(ResolvedEnrichPolicy.java:32)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:737)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:721)\n\
      \tat org.elasticsearch.xpack.esql.enrich.EnrichPolicyResolver$LookupResponse.<init>(EnrichPolicyResolver.java:333)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:63)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:25)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1471)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1458)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.InboundHandler.doHandleResponse(InboundHandler.java:418)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.InboundHandler$2.doRun(InboundHandler.java:389)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\
      \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\
      \tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\
      \tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\
      \tat java.base/java.lang.Thread.run(Thread.java:1570)\n"
  stack_trace: "org.elasticsearch.transport.TransportSerializationException: Failed\
    \ to deserialize response from handler [ContextRestoreResponseHandler[ActionListenerResponseHandler[refCounting[org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener/WrappedActionListener{org.elasticsearch.xpack.esql.execution.PlanExecutor$$Lambda/0x00007f6ec85a16b8@2149c9cd}{org.elasticsearch.xpack.esql.execution.PlanExecutor$$Lambda/0x00007f6ec85a18e8@72437f48}/org.elasticsearch.xpack.esql.session.EsqlSession$$Lambda/0x00007f6ec85dd548@5c0b27a/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@58067d8d/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@376ed0e1/org.elasticsearch.xpack.esql.core.util.ActionListeners$$Lambda/0x00007f6ec85e0820@5e3e507b/org.elasticsearch.xpack.esql.session.EsqlSession$$Lambda/0x00007f6ec85df790@5eed7aa4/org.elasticsearch.action.ActionListenerImplementations$MappedActionListener$$Lambda/0x00007f6ec81f6ad8@356c8ee1][org.elasticsearch.xpack.esql.enrich.EnrichPolicyResolver$$Lambda/0x00007f6ec86564d0@74c4b975]]]]\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.InboundHandler.doHandleResponse(InboundHandler.java:421)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.InboundHandler$2.doRun(InboundHandler.java:389)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:984)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)\n\
    \tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)\n\
    \tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)\n\
    \tat java.base/java.lang.Thread.run(Thread.java:1570)\nCaused by: java.lang.IllegalArgumentException:\
    \ Invalid EsField type [airport]\n\tat org.elasticsearch.xpack.esql.core.type.EsField.getReader(EsField.java:38)\n\
    \tat org.elasticsearch.xpack.esql.io.stream.PlanStreamInput.readEsFieldWithCache(PlanStreamInput.java:280)\n\
    \tat org.elasticsearch.xpack.esql.core.type.EsField.readFrom(EsField.java:70)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:737)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:714)\n\
    \tat org.elasticsearch.xpack.esql.enrich.ResolvedEnrichPolicy.<init>(ResolvedEnrichPolicy.java:32)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:737)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.common.io.stream.StreamInput.readMap(StreamInput.java:721)\n\
    \tat org.elasticsearch.xpack.esql.enrich.EnrichPolicyResolver$LookupResponse.<init>(EnrichPolicyResolver.java:333)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:63)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.action.ActionListenerResponseHandler.read(ActionListenerResponseHandler.java:25)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1471)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1458)\n\
    \tat org.elasticsearch.server@8.15.2-SNAPSHOT/org.elasticsearch.transport.InboundHandler.doHandleResponse(InboundHandler.java:418)\n\
    \t... 8 more\n"
status: 500

```

**Issue Reasons:**
- [8.15] 2 consecutive failures in step 8.14.3_bwc
- [8.15] 2 consecutive failures in step 8.13.4_bwc
- [8.15] 4 failures in class org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT (11.1% fail rate in 36 executions)
- [8.15] 2 failures in step 8.14.3_bwc (50.0% fail rate in 4 executions)
- [8.15] 2 failures in step 8.13.4_bwc (50.0% fail rate in 4 executions)
- [8.15] 2 failures in pipeline elasticsearch-periodic (50.0% fail rate in 4 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/docs/changelog/112985.yaml
+++ b/docs/changelog/112985.yaml
@@ -2,4 +2,5 @@ pr: 112985
 summary: "ES|QL: Fix `ResolvedEnrichPolicy` serialization (bwc) in v 8.15"
 area: ES|QL
 type: bug
-issues: []
+issues:
+ - 112968

--- a/docs/changelog/112985.yaml
+++ b/docs/changelog/112985.yaml
@@ -1,0 +1,5 @@
+pr: 112985
+summary: "ES|QL: Fix `ResolvedEnrichPolicy` serialization (bwc) in v 8.15"
+area: ES|QL
+type: bug
+issues: []

--- a/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DateEsField.java
+++ b/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DateEsField.java
@@ -30,7 +30,7 @@ public class DateEsField extends EsField {
     }
 
     @Override
-    protected void writeContent(StreamOutput out) throws IOException {
+    public void writeContent(StreamOutput out) throws IOException {
         out.writeString(getName());
         out.writeMap(getProperties(), (o, x) -> x.writeTo(out));
         out.writeBoolean(isAggregatable());

--- a/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/EsField.java
+++ b/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/EsField.java
@@ -58,7 +58,7 @@ public class EsField implements Writeable {
         this.isAlias = isAlias;
     }
 
-    protected EsField(StreamInput in) throws IOException {
+    public EsField(StreamInput in) throws IOException {
         this.name = in.readString();
         this.esDataType = DataType.readFrom(in);
         this.properties = in.readImmutableMap(EsField::readFrom);
@@ -80,7 +80,7 @@ public class EsField implements Writeable {
     /**
      * This needs to be overridden by subclasses for specific serialization
      */
-    protected void writeContent(StreamOutput out) throws IOException {
+    public void writeContent(StreamOutput out) throws IOException {
         out.writeString(name);
         out.writeString(esDataType.typeName());
         out.writeMap(properties, (o, x) -> x.writeTo(out));

--- a/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/InvalidMappedField.java
+++ b/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/InvalidMappedField.java
@@ -60,7 +60,7 @@ public class InvalidMappedField extends EsField {
     }
 
     @Override
-    protected void writeContent(StreamOutput out) throws IOException {
+    public void writeContent(StreamOutput out) throws IOException {
         out.writeString(getName());
         out.writeString(errorMessage);
         out.writeMap(getProperties(), (o, x) -> x.writeTo(out));

--- a/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/KeywordEsField.java
+++ b/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/KeywordEsField.java
@@ -70,7 +70,7 @@ public class KeywordEsField extends EsField {
     }
 
     @Override
-    protected void writeContent(StreamOutput out) throws IOException {
+    public void writeContent(StreamOutput out) throws IOException {
         out.writeString(getName());
         out.writeMap(getProperties(), (o, x) -> x.writeTo(out));
         out.writeBoolean(isAggregatable());

--- a/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/TextEsField.java
+++ b/x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/TextEsField.java
@@ -36,7 +36,7 @@ public class TextEsField extends EsField {
     }
 
     @Override
-    protected void writeContent(StreamOutput out) throws IOException {
+    public void writeContent(StreamOutput out) throws IOException {
         out.writeString(getName());
         out.writeMap(getProperties(), (o, x) -> x.writeTo(out));
         out.writeBoolean(isAggregatable());

--- a/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/ResolvedEnrichPolicy.java
+++ b/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/ResolvedEnrichPolicy.java
@@ -7,6 +7,7 @@
 
 package org.elasticsearch.xpack.esql.enrich;
 
+import org.elasticsearch.TransportVersions;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
@@ -29,8 +30,15 @@ public record ResolvedEnrichPolicy(
             in.readString(),
             in.readStringCollectionAsList(),
             in.readMap(StreamInput::readString),
-            in.readMap(EsField::readFrom)
+            in.readMap(getEsFieldReader(in))
         );
+    }
+
+    private static Reader<EsField> getEsFieldReader(StreamInput in) {
+        if (in.getTransportVersion().isPatchFrom(TransportVersions.ESQL_ATTRIBUTE_CACHED_SERIALIZATION_8_15)) {
+            return EsField::readFrom;
+        }
+        return EsField::new;
     }
 
     @Override
@@ -45,7 +53,13 @@ public record ResolvedEnrichPolicy(
              * There are lots of subtypes of ESField, but we always write the field
              * as though it were the base class.
              */
-            (o, v) -> new EsField(v.getName(), v.getDataType(), v.getProperties(), v.isAggregatable(), v.isAlias()).writeTo(o)
+            (o, v) -> {
+                if (out.getTransportVersion().isPatchFrom(TransportVersions.ESQL_ATTRIBUTE_CACHED_SERIALIZATION_8_15)) {
+                    new EsField(v.getName(), v.getDataType(), v.getProperties(), v.isAggregatable(), v.isAlias()).writeTo(o);
+                } else {
+                    new EsField(v.getName(), v.getDataType(), v.getProperties(), v.isAggregatable(), v.isAlias()).writeContent(o);
+                }
+            }
         );
     }
 }

